R/servers.R

```diff
@@ -179,13 +181,19 @@ addConnectServer <- function(url, name = NULL, certificate = NULL,
 
 #' @rdname addConnectServer
 #' @export
-addServer <- function(url, name = NULL, certificate = NULL, quiet = FALSE) {
+addServer <- function(url, name = NULL, certificate = NULL, validate = TRUE, quiet = FALSE) {
   check_string(url)
   check_name(name, allow_null = TRUE)
 
-  serverUrl <- parseHttpUrl(url)
+  if (validate) {
+    out <- validateConnectUrl(url, certificate)
```

Comment:
Sneaky indirection. `addServer -> validateConnectUrl -> findLocalServer -> addServer`.

Which means: When validating, `addServer` has already been called. Return immediately (or somehow restructure things to not be so sneaky).

Reply:
`validateConnectUrl()` doesn't call `findLocalServer()`; `validateUrl()` does. `validateUrl()` needs a better name, but we can't change it because the IDE depends on it.

Reply:
Ugh. Sorry. These names!

Reply:
I've done what I can to improve the comments and clarify what function names I can affect.